### PR TITLE
[RFC] ASoC: SOF: remove intel-specific field from snd_sof_dev{}

### PR DESCRIPTION
**Pull request by bardliao**
We don't want to keep intel-specific data in struct snd_sof_dev{} which
is on the top level of SOF. This patch do the following
1. remove "struct sof_intel_hda_dev *hda;" from snd_sof_dev{} struct
2. remove spi-specific fields from snd_sof_dev{} struct
3. add struct sof_spi_dev{} to store spi-specific data
4. add "void *hw_pdata;" into snd_sof_pdata{}. It will be
   converted to struct sof_intel_hda_dev or struct sof_spi_dev
The *hw_pdata can be used to store SoC or hardware-stuff.

Signed-off-by: Bard liao <yung-chuan.liao@linux.intel.com>

### CODE REVIEW

File: sound/soc/sof/intel/hda.c [hunk 1]
```diff
@@ -444,7 +444,7 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	hdev = devm_kzalloc(&pci->dev, sizeof(*hdev), GFP_KERNEL);
 	if (!hdev)
 		return -ENOMEM;
-	sdev->hda = hdev;
+	sdev->pdata->hw_pdata = (struct sof_intel_hda_dev *)hdev;
```
Comment (keyonjie) [inline sound/soc/sof/intel/hda.c]: sdev->pdata->hw_pdata = (void *)hdev;
?

Comment (bardliao) [reply sound/soc/sof/intel/hda.c]: Thanks @keyonjie I will check it. Probably we don't need to cast it here.

**Comment by bardliao** [comment]
@plbossart It should be able to build now. :)

**Comment by bardliao** [comment]
Of cause, thanks @plbossart

**Pull request merged**